### PR TITLE
EDSC-2300: Adds download history page

### PR DESCRIPTION
**Pull request by abbottry**

### CODE REVIEW

File: static/src/js/util/request/retrievalRequest.js [hunk 1]
```diff
@@ -16,6 +16,10 @@ export default class RetrievalRequest extends Request {
     ]
   }
 
+  all() {
```
Comment (macrouch) [inline static/src/js/util/request/retrievalRequest.js]: This file doesn't have tests

File: static/src/js/reducers/retrievalHistory.js [hunk 2]
```diff
@@ -0,0 +1,17 @@
+import { SET_RETRIEVAL_HISTORY } from '../constants/actionTypes'
+
+const initialState = []
+
+const retrievalHistoryReducer = (state = initialState, action) => {
+  switch (action.type) {
+    case SET_RETRIEVAL_HISTORY: {
```
Comment (macrouch) [inline static/src/js/reducers/retrievalHistory.js]: write a test

File: static/src/js/containers/DownloadHistoryContainer/DownloadHistoryContainer.js [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+    // No collection title exists so we'll just display how many collections were downloaded
+    return `${collections.length} ${pluralize('collection', collections.length)}`
+  }
+
+  render() {
+    const { retrievalHistory } = this.props
+
+    return (
+      <>
+        <h2 className="route-wrapper__page-heading">Download Status & History</h2>
```
Comment (macrouch) [inline static/src/js/containers/DownloadHistoryContainer/DownloadHistoryContainer.js]: This normally goes in a component, not a container. It is a simple page, but should we follow the way the rest of the application is laid out?

File: static/src/js/containers/DownloadHistoryContainer/DownloadHistoryContainer.js [hunk 4]
```diff
@@ -0,0 +1,50 @@
+import React, { Component } from 'react'
+import { connect } from 'react-redux'
+import PropTypes from 'prop-types'
+import { withRouter } from 'react-router-dom'
+
+import actions from '../../actions'
+import { DownloadHistory } from '../../components/DownloadHistory/DownloadHistory'
+
+const mapStateToProps = state => ({
+  authToken: state.authToken,
```
Comment (macrouch) [inline static/src/js/containers/DownloadHistoryContainer/DownloadHistoryContainer.js]: are authToken and portal used? I think they can be removed

File: static/src/js/containers/DownloadHistoryContainer/DownloadHistoryContainer.js (hunk 3 above)
Comment (trevorlang) [reply static/src/js/containers/DownloadHistoryContainer/DownloadHistoryContainer.js]: This is sort of what I was referring to when we need a way to wrap these routes and their subcomponents consistently, probably through a separate component. I think we can probably leave it the way it is until we sort that out 🤷‍♂

Comment (abbottry) [reply static/src/js/containers/DownloadHistoryContainer/DownloadHistoryContainer.js]: Eh, I moved it.

**Pull request merged**